react/src/components/ServiceLauncherModal.tsx

```diff
@@ -458,191 +469,237 @@ const ServiceLauncherModal: React.FC<ServiceLauncherProps> = ({
-                valuePropName="checked"
-              >
-                <Switch disabled={endpoint ? true : false}></Switch>
-              </Form.Item>
-              {/* <VFolderTableFromItem /> */}
-              {!endpoint ? (
+                  }
+                : {
+                    desiredRoutingCount: 1,
+                    ...RESOURCE_ALLOCATION_INITIAL_FORM_VALUES,
+                    ...(baiClient._config?.default_session_environment && {
+                      environments: {
+                        environment:
+                          baiClient._config?.default_session_environment,
+                      },
+                    }),
+                  }
+            }
+            requiredMark="optional"
+          >
+            {(baiClient.supports('modify-endpoint') || !endpoint) && (
+              <>
                 <Form.Item
-                  name={'vFolderName'}
-                  label={t('session.launcher.ModelStorageToMount')}
+                  label={t('modelService.ServiceName')}
+                  name="serviceName"
                   rules={[
+                    {
+                      pattern: /^(?=.{4,24}$)\w[\w.-]*\w$/,
+                      message: t('modelService.ServiceNameRule'),
+                    },
                     {
                       required: true,
                     },
                   ]}
                 >
-                  <VFolderSelect
-                    filter={(vf) =>
-                      vf.usage_mode === 'model' && vf.status === 'ready'
-                    }
-                    autoSelectDefault
-                    disabled={endpoint ? true : false}
-                  />
+                  <Input disabled={!!endpoint} />
                 </Form.Item>
-              ) : (
-                endpoint?.model && (
+                <Form.Item
+                  name="openToPublic"
+                  label={t('modelService.OpenToPublic')}
+                  valuePropName="checked"
+                >
+                  <Switch disabled={!!endpoint}></Switch>
+                </Form.Item>
+                {/* <VFolderTableFromItem /> */}
+                {!endpoint ? (
                   <Form.Item
+                    name={'vFolderName'}
                     label={t('session.launcher.ModelStorageToMount')}
-                    required
+                    rules={[
+                      {
+                        required: true,
+                      },
+                    ]}
                   >
-                    <Suspense fallback={<Skeleton.Input active />}>
-                      <VFolderLazyView uuid={endpoint?.model} />
-                    </Suspense>
+                    <VFolderSelect
+                      filter={(vf) =>
+                        vf.usage_mode === 'model' && vf.status === 'ready'
+                      }
+                      autoSelectDefault
+                      disabled={endpoint ? true : false}
```

Comment:
```suggestion
                      disabled={!!endpoint}
```
Oops this one too :)